main_app/views.py

```diff
@@ -136,10 +143,32 @@ class LogoutPageView(LogoutView):
     pass
 
 
-class UserPageView(TemplateView):
+class UserPageView(LoginRequiredMixin, TemplateView):
     template_name = 'user_page.html'
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['age'] = round((datetime.date(datetime.now()) - self.request.user.birthday).days / 365.25)
         return context
+
+
+class MustPageView(LoginRequiredMixin, TemplateView):
+    template_name = 'must_page.html'
+
+    def post(self, request, **kwargs):
```

Comment:
The proper way here is to create separate view - just http://ccbv.co.uk/projects/Django/2.2/django.views.generic.base/View/

I suggest you to have new url like /add_remove_must It will accept 2 POST params: add = true/false and game_id
In this way the code will be separated into logical pieces - Must page in separate view and endpoint that adds/remove - in separate also